homeassistant/components/mqtt/__init__.py

```diff
@@ -1182,9 +1208,9 @@ def discovery_callback(payload):
             if not payload:
                 # Empty payload: Remove component
                 _LOGGER.info("Removing component: %s", self.entity_id)
+                self._cleanup_on_remove()
+                self.hass.async_create_task(async_remove_from_registry(self))
```

Comment:
Why don't we await it ?

Reply:
Is it safe to do so?
I think it was intentional to not await directly on the line below:
https://github.com/home-assistant/core/blob/701731a9eecbed59439c96f7807a190b3743ea18/homeassistant/components/mqtt/__init__.py#L1213

Reply:
It comes from a dispatcher. I think that it's fine to do so.

Reply:
OK, changed.